src/phoebus_boundaries/phoebus_boundaries.cpp

```diff
@@ -460,82 +470,90 @@ void ProcessBoundaryConditions(parthenon::ParthenonManager &pman) {
       pman.pinput->GetOrAddString("radiation", "method", "None");
 
+  // TODO: logic for gr outflow et/
   if (typeid(PHOEBUS_GEOMETRY) == typeid(Geometry::FMKS)) {
     bool derefine_poles =
         pman.pinput->GetOrAddBoolean("coordinates", "derefine_poles", true);
     if (derefine_poles) {
       const std::string ix2_bc =
-          pman.pinput->GetOrAddString("phoebus", "ix2_bc", "outflow");
+          pman.pinput->GetOrAddString("phoebus", "ix2_bc", "gr_outflow");
       const std::string ox2_bc =
-          pman.pinput->GetOrAddString("phoebus", "ox2_bc", "outflow");
+          pman.pinput->GetOrAddString("phoebus", "ox2_bc", "gr_outflow");
       PARTHENON_REQUIRE(
           ix2_bc != "reflect" && ix2_bc != "polar" && ox2_bc != "reflect" &&
               ox2_bc != "polar",
           "Polar and Reflecting X2 BCs not supported for \"derefine_poles = true\"!");
     }
   }
+  pman.app_input->RegisterSwarmBoundaryCondition(
+      parthenon::BoundaryFace::inner_x3, "swarm_no_bc", &Boundaries::SwarmNoWorkBC);
 
   for (int d = 1; d <= 3; ++d) {
     // outer = 0 for inner face, outer = 1 for outer face
     for (int outer = 0; outer <= 1; ++outer) {
       auto &face = inner_outer[outer];
       const std::string name = face + "x" + std::to_string(d) + "_bc";
       const std::string parth_bc = pman.pinput->GetString("parthenon/mesh", name);
-      PARTHENON_REQUIRE(parth_bc == "user" || parth_bc == "periodic",
-                        "Only \"user\" and \"periodic\" allowed for parthenon/mesh/" +
-                            name);
+      // TODO: proper check here??
+      //      PARTHENON_REQUIRE(parth_bc == "user" || parth_bc == "periodic",
+      //                        "Only \"user\" and \"periodic\" allowed for
+      //                        parthenon/mesh/" +
+      //                            name);
 
-      const std::string bc = pman.pinput->GetOrAddString("phoebus", name, "outflow");
+      const std::string bc = pman.pinput->GetOrAddString("phoebus", name, "gr_outflow");
       if (bc == "reflect") {
-        pman.app_input->boundary_conditions[loc[d - 1][outer]] = reflect[d - 1][outer];
+        pman.app_input->RegisterBoundaryCondition(loc[d - 1][outer], "reflect",
+                                                  reflect[d - 1][outer]);
       } else if (bc == "polar") {
         PARTHENON_REQUIRE(d == 2, "Polar boundary conditions only supported in X2!");
-        pman.app_input->boundary_conditions[loc[d - 1][outer]] = polar[d - 1][outer];
-      } else if (bc == "outflow") {
-        pman.app_input->boundary_conditions[loc[d - 1][outer]] = outflow[d - 1][outer];
+        pman.app_input->RegisterBoundaryCondition(loc[d - 1][outer], "polar",
+                                                  polar[d - 1][outer]);
+      } else if (bc == "gr_outflow") {
+        pman.app_input->RegisterBoundaryCondition(loc[d - 1][outer], "gr_outflow",
+                                                  outflow[d - 1][outer]);
```

Comment:
No need t put these inside if statements. Just register them all. Parthenon can do the option switching.